ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -772,6 +775,183 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
     }
   }
 
+  @Override
+  public StoreBatchDeleteInfo batchDelete(List<MessageInfo> infosToDelete) throws StoreException {
+    checkStarted();
+    checkDuplicates(infosToDelete);
+    StoreBatchDeleteInfo storeBatchDeleteInfo = new StoreBatchDeleteInfo(
+        replicaId.getPartitionId(), new ArrayList<>());
+    final Timer.Context context = metrics.batchDeleteResponse.time();
+    int i = 0;
+    List<MessageInfo> filteredInfos = null;
+    try {
+      List<IndexValue> indexValuesPriorToDelete = new ArrayList<>();
+      List<IndexValue> originalPuts = new ArrayList<>();
+      List<Short> lifeVersions = new ArrayList<>();
+      Offset indexEndOffsetBeforeCheck = index.getCurrentEndOffset();
+      filteredInfos = new ArrayList<>();
+      for (MessageInfo info : infosToDelete) {
+        try {
+          IndexValue value =
+              index.findKey(info.getStoreKey(), new FileSpan(index.getStartOffset(), indexEndOffsetBeforeCheck));
+          if (value == null) {
+            throw new StoreException(
+                "Cannot delete id " + info.getStoreKey() + " because it is not present in the index",
+                StoreErrorCodes.ID_Not_Found);
+          }
+          if (!info.getStoreKey().isAccountContainerMatch(value.getAccountId(), value.getContainerId())) {
+            if (config.storeValidateAuthorization) {
+              throw new StoreException(
+                  "DELETE authorization failure. Key: " + info.getStoreKey() + "Actually accountId: "
```

Comment:
nit: Better to create the error string as a separate statement.

```
string error = "DELETE authorization failure. Key: " + info.getStoreKey() + "Actually accountId: " + value.getAccountId() + "Actually containerId: " + value.getContainerId();

if (config.storeValidateAuthorization) {
    throw new StoreException(error);
} else {
    logger.warn(error);
}
```

That way the error strings stay consistent & can evolve consistently across both places.